app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3273,13 +3289,49 @@ public void onStartDrag(final PlayQueueItemHolder viewHolder) {
     @Override // own playback listener
     @Nullable
     public MediaSource sourceOf(final PlayQueueItem item, final StreamInfo info) {
-        return (isAudioOnly ? audioResolver : videoResolver).resolve(info);
+        final MediaSource mediaSource = (isAudioOnly ? audioResolver : videoResolver)
+                .resolve(info);
+        if ((isAudioOnly ? audioResolver : videoResolver).isLiveSource()) {
+            setDefaultResolutionLimit();
+        }
+        return mediaSource;
     }
 
     public void disablePreloadingOfCurrentTrack() {
         loadController.disablePreloadingOfCurrentTrack();
     }
 
+    private void setDefaultResolutionLimit() {
+        final String defaultResolution;
+
+        if (popupPlayerSelected()) {
+            defaultResolution = computeDefaultResolution(context,
+                    R.string.default_popup_resolution_key,
+                    R.string.default_popup_resolution_value);
+        } else {
+            defaultResolution = computeDefaultResolution(context,
+                    R.string.default_resolution_key, R.string.default_resolution_value);
+        }
+
+        // Best resolution set: not needed to set a maximum resolution
+        if (!defaultResolution.equals(context.getString(R.string.best_resolution_key))) {
+            final String[] defaultResolutionArray = defaultResolution.split("p");
+            final int defaultHeight = Integer.parseInt(defaultResolutionArray[0]);
+            final int defaultFrameRate = defaultResolutionArray.length == 2
+                    ? Integer.parseInt(defaultResolutionArray[1]) : -1;
+
+            final DefaultTrackSelector.Parameters selectionParameters = trackSelector
+                    .getParameters();
+            final DefaultTrackSelector.ParametersBuilder builder = selectionParameters.buildUpon()
+                    .clearSelectionOverrides();
+
+            builder.setMaxVideoSize(Integer.MAX_VALUE, defaultHeight);
+            builder.setMaxVideoFrameRate(defaultFrameRate != -1 ? defaultFrameRate : 30);
```

Comment:
And remove the variable above
```suggestion
            builder.setMaxVideoFrameRate(defaultResolutionArray.length == 2
                    ? Integer.parseInt(defaultResolutionArray[1]) : 30);
```